server/src/main/proto/spine/server/dispatch/dispatching.proto

```diff
@@ -118,17 +118,25 @@ message Success {
 }
 
 // Events produced during a signal handling.
+//
+// Spine supports the handler methods returning `Optional` values,
+// so there may be no produced events.
+//
 message ProducedEvents {
     option (is).java_type = "ProducedEventsMixin";
 
-    repeated core.Event event = 1 [(required) = true, (validate) = true];
+    repeated core.Event event = 1 [(validate) = true];
 }
 
 // Commands produced during a signal handling.
+//
+// Spine supports the handler methods returning `Optional` values,
+// so there may be no produced commands.
+//
 message ProducedCommands {
     option (is).java_type = "ProducedCommandsMixin";
 
-    repeated core.Command command = 1 [(required) = true, (validate) = true];
```

Comment:
Why are validation rules in this file changed?

Reply:
@dmdashenkov that's because there may be no produced commands. In our code, we allow to return `Optional<Cmd>` in the commanding-splitting methods. Previously, the validation of instances of this type was never performed. Now, as the validation code is executed upon `build()`, it is always validated, and lots of tests fail.